docs/content/sips/013-sqlite.md

```diff
@@ -0,0 +1,165 @@
+* Is there really a need for query *and* execute functions since at the end of the day, they are basically equivalent?
+
+#### Database migrations
+
+Database tables typically require some sort of configuration in the form of database migrations to get table schemas into the correct state. To begin with a command line option supplied to `spin up` will be available for running SQL statements on start up (i.e., `--sqlite-migration "CREATE TABLE users..."`). It will be up to the user to provide itempotent statements such that running them multiple times does not produce unexpected results.
+
+However, such an approach (while useful) is likely to not be sufficient for more advanced use cases. We could require the user to ensure that the database is in the correct state each time the trigger handler function is run. However, there are a few issues with this:
+* Schema tracking schemes (e.g., a "migrations" table) themselves require some sort of bootstrap step.
+* This goes against the design principle of keeping components handler functions simple and single purpose.
+
+There are several possible ways to address this issue such as:
+* Some mechanism for running spin components before others where the component receives the current schema version and decides whether or not to perform migrations. 
+* The spin component could expose a current schema version as an exported value type so that an exported function would not need to called. If the exported schema version does not match the current schema version, an exported migrate function then gets called.
+* A spin component that gets called just after pre-initialization finishes. Similarly, this component would expose a schema version and have an exported migration function called when the exported schema version does not match the current schema version.
+* Configuration option in spin.toml manifest for running arbitrary SQL instructions on start up (e.g., `sqlite.migration = "CREATE TABLE users..."`)
+
+It should be noted that many of these options are not mutually exclusive and we could introduce more than one (perhaps starting with one option that will mostly be replaced later with a more generalized approach).
+
+For now, we punt on this question and only provide a mechanism for running SQL statements on start up.
+
+#### Implementation requirements
+
+**TODO**: Open questions:
+* Assumed sqlite version?
+  * Semantics may change slightly depending on the sqlite version. It's unlikely that we'll be able to match the exact versions between whatever sqlite implementation spin users, Fermyon Cloud, and the user (if they decide to create their own databases manually). Having some guidance on which versions are expected to work might make it easier to guide the user down the right path.
+* Capacity limits? The following are different capacities we might want to control:
```

Comment:
This feels like something we probably don't need to be concerned about in local Spin.  Fermyon Cloud or other hosts may want to place limits, but I don't think we need to plan for that here (except maybe assigning an error code for them).

The main thing I can think of locally is if there could be problems with the size of a query result set, which brings us back round to the list vs cursor vs hybrid issue all over again...

Reply:
Perhaps we should give some sort of *minimum* requirement that we expect every host environment running spin applications to conform to. The [key-value SIP does this](https://github.com/fermyon/spin/blob/608b5f4b11914109e84f9c950cba6db5b50ce512/docs/content/sips/010-key-value.md?plain=1#L123-L125) for that feature.

Reply:
I like that plan!

Reply:
Done!